actors/common/src/main/java/org/sunbird/common/quartz/scheduler/ShadowUserMigrationScheduler.java

```diff
@@ -0,0 +1,312 @@
+        }
+        return bulkMigrationUser;
+    }
+
+    private List<MigrationUser> getMigrationUserAsList(BulkMigrationUser bulkMigrationUser) {
+        List<MigrationUser> migrationUserList = new ArrayList<>();
+        try {
+            String decryptedData = decryptionService.decryptData(bulkMigrationUser.getData());
+            migrationUserList = mapper.readValue(decryptedData, new TypeReference<List<MigrationUser>>() {
+            });
+        } catch (Exception e) {
+            e.printStackTrace();
+            ProjectLogger.log("ShadowUserMigrationScheduler:getMigrationUserAsList:error occurred while converting map to pojo".concat(e.getMessage() + ""), LoggerEnum.ERROR.name());
+        }
+        return migrationUserList;
+    }
+
+    private Map<String, Object> getShadowExistingUserDetails(String channel, String userExtId) {
+        Map<String, Object> propertiesMap = new HashMap<>();
+        propertiesMap.put(JsonKey.CHANNEL, channel);
+        propertiesMap.put("userExtId", userExtId);
+        Map<String, Object> result = new HashMap<>();
+        Response response = cassandraOperation.getRecordsByProperties(JsonKey.SUNBIRD, JsonKey.SHADOW_USER, propertiesMap);
+        if (!((List) response.getResult().get(JsonKey.RESPONSE)).isEmpty()) {
+            result = ((Map) ((List) response.getResult().get(JsonKey.RESPONSE)).get(0));
+        }
+        return result;
+    }
+
+    /**
+     * this method will prepare the shawdow user object and insert the record into shawdow_user table
+     * @param createdBy
+     * @param processId
+     * @param migrationUser
+     */
+    private void insertShadowUserToDb(String createdBy,String processId, MigrationUser migrationUser) {
+        ShadowUser shadowUser = new ShadowUser.ShadowUserBuilder()
+                .setUserExtId(migrationUser.getUserExternalId())
+                .setOrgExtId(migrationUser.getOrgExternalId())
+                .setEmail(migrationUser.getEmail())
+                .setPhone(migrationUser.getPhone())
+                .setUserId(null)
+                .setAddedBy(createdBy)
+                .setUserIds(null)
+                .setChannel(migrationUser.getChannel())
+                .setName(migrationUser.getName())
+                .setProcessId(processId)
+                .setClaimedOn(null)
+                .setClaimStatus(ClaimStatus.UNCLAIMED.getValue())
+                .setUserStatus(getInputStatus(migrationUser.getInputStatus()))
+                .setUpdatedOn(null)
+                .build();
+        Map<String, Object> dbMap = mapper.convertValue(shadowUser, Map.class);
+        dbMap.put(JsonKey.CREATED_ON, new Timestamp(System.currentTimeMillis()));
+        if(!isOrgExternalIdValid(migrationUser)) {
+            dbMap.put(JsonKey.CLAIM_STATUS,ClaimStatus.REJECTED.getValue());
```

Comment:
Check PRD for REJECTED status.

Reply:
@manzarul  this rejected should be failed right?